### PR TITLE
[ci/release] Add schema validation for release test config

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

To avoid breakage like in #22905, this PR adds schema validation to the release test package.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/BUILD [hunk 1]
```diff
@@ -58,42 +58,49 @@ py_test(
 
 py_test(
     name = "test_alerts",
-    tags = ["team:ml", "release_unit"],
+    tags = ["team:ci", "release_unit"],
```
Comment (krfricke) [inline release/BUILD:61]: I know there's no "CI team", but team:ml seemed wrong, too, so let's use that for now.

File: release/ray_release/schema.json [hunk 2]
```diff
@@ -0,0 +1,174 @@
+{
```
Comment (simon-mo) [inline release/ray_release/schema.json:1]: any chance we can define this in pydantic instead? So we don't need to hand write the json schema as the types evolves.

**Review by rkooo567 — APPROVED** [review]
Also prefer pydantic, but not a strong blocker for me

**Comment by krfricke** [comment]
Happy to switch to pydantic eventually. However, this will require rewriting more of the code base (`Test` will move from a dict to a pydantic model), so let's do this in a follow-up PR. I'll merge this now to have the validation in place and put pydantic on my backlog as P1.

**Pull request merged**